docs/latest/design/egctl.md

```diff
@@ -0,0 +1,64 @@
+# Introduce egctl
+
+## Motivation
+
+EG should provide a command line tool with following capabilities:
+
+- One-stop installation/uninstallation experience
+- Collect configuration form data plane
+- Change envoy logging configuration on the fly
+- Analyse system configuration
+
+This tool is named `egctl`.
+
+## Syntax
+
+Use the following syntax to run `egctl` commands from your terminal window:
+
+```console
+egctl [command] [subcommand] [name] [flags]
+```
+
+where `command`, `name`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources,
+  for example `install`, `uninstall`, `proxy-config`, `version`.
+
+* `name`: Specifies the name of the specified pod. 
+
+* `flags`: Specifies optional flags. For example, you can use the `-f` or `--file` flags to specify the values for installing.
+
+If you need help, run `kubectl help` from the terminal window.
+
+## Operation
+
+The following table includes short descriptions and the general syntax for all the `egctl` operations:
+
+| Operation      | Syntax                          | Description                                               |
+| -------------- |---------------------------------|-----------------------------------------------------------|
+| `version`      | `egctl version`                 | Prints out build version information.                     |
+| `install`      | `egctl install -f FILENAME`     | Install or reconfigure EG on a cluster.                   |
+| `uninstall`    | `egctl uninstall`               | Uninstall EG from a cluster                               |
+| `proxy-config` | `egctl proxy-config SUBCOMMAND` | Retrieve information about proxy configuration from envoy |
```

Comment:
prefer having two subcommands here - `envoy-gateway` and `envoy-proxy` to add more features within each (data plane and control plane)

Reply:
thinking again, should this be replaced with `collect` instead, so
```
egctl collect all
```
wil collect everything (logs and config) for data plane and control plane

```
egctl collect config --envoy-proxy
```

will collect config associated with envoy proxy

Reply:
what should `egctl` collect from envoy-gateway, is there any debug endpoint on `envoy-gateway`?

Reply:
nothing today, but as you hinted, will need a similar admin endpoint like envoy to toggle log level, get EG config, etc, so we can start building with keeping that in mind, but it shouldnt block this proposal

Reply:
my original think was adding a subcommand like `egctl gateway-config` or else when we have that debug endpoint.

Reply:
gentle ping @arkodg

Reply:
yeah so the discussion is around what the parent noun should be , imho top level should be "what the user is trying to achieve" - like `install` or `collect` (info) or `analyze` 
@danehans @AliceProxy @youngnick @Xunzhuo @skriss can you share your thoughts as well

Reply:
> yeah so the discussion is around what the parent noun should be , imho top level should be "what the user is trying to achieve" - like install or collect (info) or analyze

Agreed. If the parent noun can be used for both control and data planes, then that noun should expose `control-plane` and `data-plane` flags are exposed to filter the action. See https://github.com/envoyproxy/gateway/pull/797/files#r1050247359 for a detailed example.